packages/flutter/test/widgets/autocomplete_test.dart

```diff
@@ -421,6 +421,130 @@ void main() {
+      final Offset optionsViewCenter = tester.getCenter(find.byType(Placeholder));
+      expect(fieldBottomLeft, offsetMoreOrLessEquals(optionsViewTopLeft));
+      expect(optionsViewCenter.dy, greaterThan(optionsViewTopLeft.dy));
+    });
+
+    testWidgets('down: open downward', (WidgetTester tester) async {
+      await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: Center(
+                child: RawAutocomplete<String>(
+                  optionsViewOpenDirection: OptionsViewOpenDirection.down, // ignore: avoid_redundant_argument_values
+                  optionsBuilder: (_) => <String>['a'],
+                  fieldViewBuilder: (_, TextEditingController controller, FocusNode focusNode, __) {
+                    return TextField(controller: controller, focusNode: focusNode);
+                  },
+                  optionsViewBuilder: (_, __, ___) => const Placeholder())))));
+      await tester.showKeyboard(find.byType(TextField));
+      final Offset fieldBottomLeft = tester.getBottomLeft(find.byType(TextField));
+      final Offset optionsViewTopLeft = tester.getTopLeft(find.byType(Placeholder));
+      final Offset optionsViewCenter = tester.getCenter(find.byType(Placeholder));
+      expect(fieldBottomLeft, offsetMoreOrLessEquals(optionsViewTopLeft));
+      expect(optionsViewCenter.dy, greaterThan(optionsViewTopLeft.dy));
+    });
+
+    testWidgets('up: open upward', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Center(
+              child: RawAutocomplete<String>(
+                optionsViewOpenDirection: OptionsViewOpenDirection.up,
+                optionsBuilder: (_) => <String>['a'],
+                fieldViewBuilder: (_, TextEditingController controller, FocusNode focusNode, __) {
+                  return TextField(controller: controller, focusNode: focusNode);
+                },
+                optionsViewBuilder: (_, __, ___) => const Placeholder())))));
+      await tester.showKeyboard(find.byType(TextField));
+      final Offset fieldTopLeft = tester.getTopLeft(find.byType(TextField));
+      final Offset optionsViewBottomLeft = tester.getBottomLeft(find.byType(Placeholder));
+      final Offset optionsViewCenter = tester.getCenter(find.byType(Placeholder));
+      expect(fieldTopLeft, offsetMoreOrLessEquals(optionsViewBottomLeft));
+      expect(optionsViewCenter.dy, lessThan(optionsViewBottomLeft.dy));
+    });
+
+    group('fieldViewBuilder not passed', () {
+      // Find the [SizedBox.shrink] that substitutes for a text field when
+      // [RawAutocomplete.fieldViewBuilder] is not passed.
+      final Finder sizedBoxShrinkFinder = find.descendant(
+        of: find.byType(RawAutocomplete<String>),
+        matching: find.byWidgetPredicate((Widget widget) {
+          return widget is SizedBox && widget.height == 0.0 && widget.width == 0.0;
```

Comment:
Hmm, this feels a bit like it's probing into the internal details of the RawAutocomplete.

How about instead passing a key to the RawAutocomplete, like `key: autocompleteKey`, and then saying `find.byKey(autocompleteKey)`? I think that will cover what this is really getting at, which is that the options view is aligned in such-and-such a way against the box where the RawAutocomplete widget itself is laid out.

Reply:
The `SizedBox.shrink()` is explicitly part of `RawAutocomplete`'s API (as mentioned on [`.textEditingController`](https://api.flutter.dev/flutter/widgets/RawAutocomplete/textEditingController.html) and [`.focusNode`](https://api.flutter.dev/flutter/widgets/RawAutocomplete/focusNode.html) before this branch, and now in this branch on `fieldViewBuilder`), so I think this finder doesn't overstep into `RawAutocomplete`'s internal details.

But your alternative wouldn't be wrong, since I think it's assumed that the `RawAutocomplete` widget itself will be laid out to match the `SizedBox.shrink()`.

Reply:
Yeah. It is part of the API, but it feels like it's a different part of the API from the one this test is about.